app/helpers/categories_helper.rb

```diff
@@ -0,0 +1,27 @@
+module CategoriesHelper
+  def parent_categories
+    Category.parent_categories
+  end
+
+  def featured_products max = Settings.length.featured_products.max
+    Product.featured_products max
+  end
+
+  def to_filters_hash products
+    products.each_with_object({}) do |e, h|
+      cat = e.category
+      # h[cat.id] = cat.title
+      h[cat.parent.id] = cat.parent.title unless cat.parent.nil?
```

Comment:
Nên dùng khẳng định sẽ tường minh hơn
```suggestion
  end

  def to_filters_hash products
    products.each_with_object({}) do |e, h|
      cat = e.category
      # h[cat.id] = cat.title
      h[cat.parent.id] = cat.parent.title if cat.parent.present?
```